### PR TITLE
Documentation and miscelaneous fixes

### PR DESCRIPTION
**Pull request by GiudGiud**
@roystgnr if you could please check the file IO list for accuracy

### CODE REVIEW

**Comment by GiudGiud** [comment]
Removing the commit fixing linker/compiler flags, in favor of getting this in sooner

File: modules/external_petsc_solver/doc/content/source/timesteppers/ExternalPetscTimeStepper.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
 #  External PETSc time stepper
+
 This time stepper is used to query dt from PETSc TS to synchronize the time step size
 between the wrapper app and the external PETSc solver.
+
+The list of PETSc time steppers / integrators can be found on
+[this summary](https://petsc.org/release/overview/integrator_table/#integrator-table).
+
+This object expects the external PETSc solver to be using a PETSc TS (PETSc time stepper)
+and retrieves the time step from this the TS.
```
Comment (lindsayad) [inline modules/external_petsc_solver/doc/content/source/timesteppers/ExternalPetscTimeStepper.md:10]: Isn't this pretty clear from lines 3 and 4?

Comment (GiudGiud) [reply modules/external_petsc_solver/doc/content/source/timesteppers/ExternalPetscTimeStepper.md]: you're right. Deleted

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review!

**Comment by roystgnr** [comment]
File I/O list was all correct; only thing I'd have added is that for most of those formats (I think Nemesis and VTK might be exceptions?) libMesh will handle gzipped files (via a library in the IO pipeline) or bzipped/xzipped files (via shell calls) automatically.

**Comment by GiudGiud** [comment]
Awesome thank you for checking!

**Pull request merged**